### DISCUSSION TITLE
Content Picker title not showing when act as drop down on differ content type

### DISCUSSION THREAD

**Discussion by saleemhaq** [OrchardCMS/OrchardCore#11108] [Q&A]
![image](https://user-images.githubusercontent.com/5202291/151813388-cb7d8a09-e519-4fda-bb9d-ef86a940a1e9.png)

![image](https://user-images.githubusercontent.com/5202291/151842429-2c614578-2912-4b88-88f3-896727056e43.png)

I created new post and link like that:

still i got the same contentid instead of title:

![image](https://user-images.githubusercontent.com/5202291/151842508-79638439-2770-4e0e-969d-43bd02e45057.png)

I used the razor pages anyone please help me on above please look into this: https://github.com/OrchardCMS/OrchardCore/discussions/11055

**Comment by Skrypt** [comment]
I'm not sure about your request but in the third screenshot above you are displaying code that should render a BlogPost while you are talking about an IndustryNews Content Type. Here, I think there is a mistake in the code. So here if you had a Content Picker on the BlogPost Content Type that allows you to select a IndustryNews content item you would do something like : 

```C#
var selectedIndustryNewsContentItemId = blogPost.Content.YourContentPickerFieldName.ContentItemIds[0];
```

Then you would need to retrieve the Content Item related to that ContentItemId from the Content Picker

```C#
va industryNews = await Orchard.GetContentItemByIdAsync((string)blogPost.Content.YourContentPickerFieldName.ContentItemIds[0]);
```

**Comment by saleemhaq** [comment]
I need a title instead of content item I'd I.e Industry News need to
display?

**Reply by Skrypt** [reply]
The example above should get you the industryNews content item. Then use it to display its TitlePart.

**Comment by saleemhaq** [comment]
Can u please share with me the example I unable to achieved.

**Reply by saleemhaq** [reply]
I tried both of the way below but unable to achieve please look on below:
https://github.com/OrchardCMS/OrchardCore/discussions/11055
https://github.com/OrchardCMS/OrchardCore/discussions/11049

Please help i stuck from my last week. @Skrypt

**Reply by Skrypt** [reply]
Ok, so what you are trying to achieve is to use a Content Picker field content item to generate the Title of your BlogPost for example.
So, from what I'm seeing you are trying to use Liquid in your TitlePart of your BlogPost TitlePart Pattern.

```liquid
{{ ContentItem.Content.Industry.Name.Text }}
```

This is wrong because "Industry" here should be the name of your Content Picker Field.

```liquid
{{ ContentItem.Content.BlogPost.CategorySelection.ContentItemIds.first }}
```

This will get you the first ContentItemId from that ContentPicker Field. Though what you want is the TitlePart.Text of that Content Picker selected content item.

As documented here : https://docs.orchardcore.net/en/latest/docs/reference/modules/Contents/#loading-from-a-content-item-id
From Liquid you can retrieve a Content Item by using this helper method.

```liquid
{% assign selected_industry_news = Content.ContentItemId["417qsjrgv97e74wvp149h4da53"] %}
```

Where you would replace the UID by your selected ContentItemId.
So : 

```liquid
{% assign selected_industry_news = Content.ContentItemId[ContentItem.Content.BlogPost.CategorySelection.ContentItemIds.first] %}
```

**Reply by Skrypt** [reply]
![image](https://user-images.githubusercontent.com/3228637/151926610-29f03c72-113f-4ea7-95ba-eeb18e6b77a2.png)

And here is the result : 

![image](https://user-images.githubusercontent.com/3228637/151926660-fc2f5508-8e5d-4dfd-beaa-47325d4336b0.png)

**Comment by saleemhaq** [comment]
I used in razor syntax I still get the content item Id instaead of title
part?

**Reply by Skrypt** [reply]
Show me the Razor template that does that. It's really complex to debug your issue without any context. Paste some code so that I can see the entire context of what's happening.

**Reply by Skrypt** [reply]
Else you have that same helper method here for Razor templates : https://docs.orchardcore.net/en/latest/docs/reference/core/Razor/

GetContentItemByIdAsync

**Comment by ns8482e** [comment]
@saleemhaq  I see you are using liquid syntax incorrectly in TitlePart, for `IndustryNews` if you don't have any field named `Name` and any part named `Industry` how could it will find  `ContentItem.Content.Industry.Name.Text`? 
The syntax should follow this naming `ContentItem.Content.[ImpicitPartName].[FieldName].Text`

`[ImpicitPartName]` = Technical name of your content type - which also an implicit part of content item in your case it's `IndustryNews`
`[FieldName]` = Name of the field defined on your content item  - in your case you don't have any field

Because it generates empty - the default value assigned to CotenteItem.DisplayText will be content item id.

If you don't need field then simply enable "Render Title" and set title as editable in setting

Also when you use content picker, in another content item, it will only store ids and you have to query as suggested by @Skrypt  
```cs
var selectedIndustryNewsContentItemId = blogPost.Content.YourContentPickerFieldName.ContentItemIds[0] as string;
va industryNews = await Orchard.GetContentItemByIdAsync(selectedIndustryNewsContentItemId);

// here is the title
var title - industryNews.DisplayText
```
